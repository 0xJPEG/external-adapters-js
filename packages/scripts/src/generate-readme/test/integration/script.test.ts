import child_process from 'child_process'
import fs from 'fs'

describe('readme generation script', () => {
  describe('when generating readme for readme-test-adapter', () => {
    it('should successfully generate readme', async () => {
      const pathToScript = 'packages/scripts/src/generate-readme/'
      const pathToAdapter = pathToScript + 'readme-test-adapter/'
      const generatedReadmePath = pathToAdapter + 'README.md'

      await child_process.exec(`yarn generate:readme ${pathToAdapter}`)

      expect(fs.readFileSync(generatedReadmePath).toString()).toMatchSnapshot()
    })
  })
})
