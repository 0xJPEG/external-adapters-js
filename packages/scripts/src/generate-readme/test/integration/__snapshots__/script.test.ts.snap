// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readme generation script when generating readme for readme-test-adapter should successfully generate readme 1`] = `
"# @chainlink/readme-test-adapter

Version: 1.2.3

## Environment Variables

| Required? |     Name     |  Type  |       Options       |            Default             |
| :-------: | :----------: | :----: | :-----------------: | :----------------------------: |
|           | API_ENDPOINT | string |                     | https://test.api.endpoint.link |
|           |   CHAIN_ID   | number |                     |               1                |
|           |     MODE     | string | live, sandbox, test |            sandbox             |
|    ✅     | PRIVATE_KEY  | string |                     |                                |
|    ✅     |   RPC_URL    | string |                     |                                |

---

## Input Parameters

| Required? |   Name   |  Type  |                                         Options                                          | Default |
| :-------: | :------: | :----: | :--------------------------------------------------------------------------------------: | :-----: |
|           | endpoint | string | [balance](#balance-endpoint), [marketcap](#marketcap-endpoint), [price](#price-endpoint) | balance |

---

## Balance Endpoint

\`balance\` is the only supported name for this endpoint.

### Input Params

| Required? |     Name      | Type | Options | Default |
| :-------: | :-----------: | :--: | :-----: | :-----: |
|           |   dataPath    |      |         |         |
|           | confirmations |      |         |         |
|           |   addresses   |      |         |         |

### Examples

Request:

\`\`\`json
{
  \\"id\\": \\"1\\",
  \\"data\\": {
    \\"endpoint\\": \\"balance\\",
    \\"dataPath\\": \\"addresses\\",
    \\"addresses\\": [
      {
        \\"address\\": \\"3EyjZ6CtEZEKyc719NZMyWaJpJG5jsVJL1\\"
      }
    ]
  }
}
\`\`\`

Response:

\`\`\`json
{
  \\"result\\": [
    {
      \\"address\\": \\"3EyjZ6CtEZEKyc719NZMyWaJpJG5jsVJL1\\",
      \\"chain\\": \\"mainnet\\",
      \\"coin\\": \\"btc\\",
      \\"balance\\": \\"2188\\"
    }
  ]
}
\`\`\`

## Marketcap Endpoint

Supported names for this endpoint are: \`marketcap\`, \`mc\`.

### Input Params

| Required? |         Name          | Type |      Options      | Default |
| :-------: | :-------------------: | :--: | :---------------: | :-----: |
|    ✅     |         base          |      | base, from, coin  |         |
|    ✅     |         quote         |      | quote, to, market |         |
|    ✅     |        coinid         |      |                   |         |
|           |      resultPath       |      |                   |         |
|           | referenceCurrencyUuid |      |                   |         |

### Examples

Request:

\`\`\`json
{
  \\"id\\": \\"1\\",
  \\"data\\": {
    \\"endpoint\\": \\"marketcap\\",
    \\"base\\": \\"BTC\\",
    \\"quote\\": \\"USD\\",
    \\"coinid\\": \\"Bitcoin\\",
    \\"resultPath\\": \\"marketcap\\"
  }
}
\`\`\`

Response:

\`\`\`json
{
  \\"result\\": 1000000000
}
\`\`\`

Request:

\`\`\`json
{
  \\"id\\": \\"1\\",
  \\"data\\": {
    \\"endpoint\\": \\"mc\\",
    \\"base\\": \\"BTC\\",
    \\"quote\\": \\"USD\\",
    \\"coinid\\": \\"Bitcoin\\",
    \\"resultPath\\": \\"marketcap\\"
  }
}
\`\`\`

Response:

\`\`\`json
{
  \\"result\\": 1000000000
}
\`\`\`

## Price Endpoint

Supported names for this endpoint are: \`price\`, \`convert\`.

### Input Params

| Required? |    Name    | Type |      Options      | Default |
| :-------: | :--------: | :--: | :---------------: | :-----: |
|    ✅     |    base    |      | base, from, coin  |         |
|    ✅     |   quote    |      | quote, to, market |         |
|           |   amount   |      |                   |         |
|           | resultPath |      |                   |         |

### Examples

Request:

\`\`\`json
{
  \\"id\\": \\"1\\",
  \\"data\\": {
    \\"endpoint\\": \\"price\\",
    \\"base\\": \\"BTC\\",
    \\"quote\\": \\"USD\\",
    \\"resultPath\\": \\"price\\"
  }
}
\`\`\`

Response:

\`\`\`json
{
  \\"result\\": 123456
}
\`\`\`

Request:

\`\`\`json
{
  \\"id\\": \\"1\\",
  \\"data\\": {
    \\"endpoint\\": \\"convert\\",
    \\"base\\": \\"BTC\\",
    \\"quote\\": \\"USD\\",
    \\"resultPath\\": \\"price\\"
  }
}
\`\`\`

Response:

\`\`\`json
{
  \\"result\\": 123456
}
\`\`\`
"
`;
